blacktie/tx/src/test/cpp/TestTransactions.cxx

```diff
@@ -527,11 +527,14 @@ void TestTransactions::test_wait_for_recovery()
 	(void) dummy_rm_add_fault(fault2);
 	(void) dummy_rm_add_fault(fault3);
 	BT_ASSERT_EQUAL(TX_OK, tx_open());
-	while ((rcCnt2 == 0 || (nrecs = count_log_records()) == nrecs1) && nsecs != 0) {
-		if (nsecs-- % 10 == 0) {
-			btlogger("TestTransactions::test_run_recovery sleeping for %d seconds", nsecs);
+	nrecs = count_log_records();
+	if(nrecs > 0 ) {
+		while ((rcCnt2 == 0 || (nrecs = count_log_records()) == nrecs1) && nsecs != 0) {
```

Comment:
This test is waiting for recovery to happen. The test for recovery is the negation of the first condition:
    (rcCnt2 == 0 || (nrecs = count_log_records()) == nrecs1)
I think the intent of the second condition (ie nsecs != 0) is to only wait for a maximum of nsecs seconds before giving up. But the way the if condition is written means that the loop will always wait for nsecs but the loop should terminate as soon as recovery is detected (ie when the first condition is false). Why not something similar to:

		const int SLEEP_INTERVAL = 10;
		
		for (int i = 0; i < nsecs; i += SLEEP_INTERVAL) {
			if (rcCnt2 == 0 || (nrecs = count_log_records()) == nrecs1)
			    break;
			btlogger("TestTransactions::test_run_recovery sleeping for another %d seconds", SLEEP_INTERVAL);
			doSix(SLEEP_INTERVAL)
		}